import-service/handler.js

```diff
@@ -0,0 +1,100 @@
+const AWS = require('aws-sdk');
+const csv = require('csv-parser');
+
+const BUCKET = 'product-service.imported-products';
+
+module.exports.importProductsFile = async function(event) {
+  try {
+    const s3 = new AWS.S3({ region: 'eu-west-1' });
+    const filename = event['queryStringParameters']['name'] || 'products';
+    const productsFilePath = `uploaded/${filename}.csv`;
+
+    const params = {
+      Bucket: BUCKET,
+      Key: productsFilePath,
+      Expires: 60,
+      ContentType: 'text/csv',
+    };
+
+    const uploadURL = await new Promise((resolve, reject) => {
+      s3.getSignedUrl('putObject', params, function (err, url) {
+        if (err) {
+          reject(err)
+        }
+        resolve(url)
+      })
+    })
+
+    return {
+      statusCode: 200,
+      headers: {
+        'Access-Control-Allow-Origin': '*',
+        'Access-Control-Allow-Credentials': true,
+      },
+      body: JSON.stringify({ uploadURL }),
+    }
+  } catch (e) {
+    return {
+      statusCode: 500,
+      headers: {
+        'Access-Control-Allow-Origin': '*',
+        'Access-Control-Allow-Credentials': true,
+      },
+      body: JSON.stringify({ error: e }),
+    };
+  }
+};
+
+module.exports.parseProductsFile = async function(event) {
+  const s3 = new AWS.S3({ region: 'eu-west-1' });
+
+  try {
+    await Promise.all(event.Records.map(record => {
+      console.log('Record key: ', record.s3.object.key);
+      const params = {
+        Bucket: BUCKET,
+        Key: record.s3.object.key,
+      }
+
+      const results = [];
+
+      const readStream = s3.getObject(params).createReadStream();
+      readStream.pipe(csv()).on('data', (data) => results.push(data));
+
+      const parsedParams = {
+        Bucket: BUCKET,
+        Key: record.s3.object.key.replace('uploaded', 'parsed'),
+        Body: readStream
+      };
+
+      return new Promise((resolve, reject) => {
+        s3.upload(parsedParams, (err, data) => {
```

Comment:
please take a look at `.promise` method. We can reduce code like `new Promise(...`

documentation - https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/using-promises.html